sql/plan.go

```diff
@@ -186,16 +197,13 @@ func (p *planner) getAliasedTableDesc(n parser.TableExpr, allowCache bool) (*Tab
 	if !ok {
 		return nil, util.Errorf("TODO(pmattis): unsupported FROM: %s", n)
 	}
-	var desc *TableDescriptor
-	var err error
-	if allowCache {
-		desc, _, err = p.getCachedTableDesc(table)
-	} else {
-		desc, err = p.getTableDesc(table)
-	}
+	leaseDesc, err := p.getTableLease(table)
 	if err != nil {
 		return nil, err
 	}
+	// Make a copy of the leased descriptor so that the caller can mutate it.
+	desc := &TableDescriptor{}
```

Comment:
Yeah, `proto.Clone` is safer. I've moved this creation of a copy into `getTableLease`.